src/cryptography/hazmat/primitives/asymmetric/dh.py

```diff
@@ -87,9 +87,6 @@ def __init__(self, p, g, q=None):
         if q is not None and not isinstance(q, six.integer_types):
             raise TypeError("q must be integer or None")
 
-        if q is None and g not in (2, 5):
-            raise ValueError("DH generator must be 2 or 5")
```

Comment:
It is, in fact, worse than DH_check since DH_check allows 2, 3, and 5. Presumably scapy wants to convert Numbers into an instance of the object though, correct? So we need to potentially remove this check and also either remove or modify DH_check?